src/App.tsx

```diff
@@ -1,111 +1,321 @@
+    />
+  );
+}
+
+function ResultCard({
+  nodeName,
+  datasetName,
+  datasetTotalSubjects,
+  numMatchingSubjects,
+  imageModals,
+}: {
+  nodeName: string;
+  datasetName: string;
+  datasetTotalSubjects: number;
+  numMatchingSubjects: number;
+  imageModals: string[];
+}) {
   return (
     <Card>
       <CardContent>
-        <Grid container wrap='nowrap' justifyContent='space-between' alignItems='center' spacing={4}>
-          <Grid item>
+        <div className="grid grid-cols-12 items-center gap-2">
+          <div className="col-end-1">
             <Checkbox />
-          </Grid>
-          <Grid item xs className='dataset-name-etc-container'>
-          <Grid container direction='column' alignItems='flex-start'>
-            <Tooltip title={
-              <Typography variant='body1'>
+          </div>
+          <div className="col-span-10 col-start-1">
+            <Tooltip
+              title={<Typography variant="body1">{datasetName}</Typography>}
+              placement="top"
+              TransitionComponent={Zoom}
+              TransitionProps={{ timeout: 500 }}
+              enterDelay={500}
+            >
+              <Typography variant="h5" className="dataset-name">
                 {datasetName}
               </Typography>
-            } 
-            placement='top' 
-            TransitionComponent={Zoom} 
-            TransitionProps={{ timeout: 500 }}
-            enterDelay={500}
-            >
-              <Typography variant="h5" className='dataset-name'>{datasetName}</Typography>
             </Tooltip>
             <Typography variant="subtitle1">from {nodeName}</Typography>
-            <Typography variant='subtitle2'>{numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects</Typography>
-          </Grid>
-          </Grid>
-          <Grid item>
+            <Typography variant="subtitle2">
+              {numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects
+            </Typography>
+          </div>
+          <div className="col-span-2 justify-self-end">
             <ButtonGroup>
               {imageModals.sort().map((modal) => (
                 <Button key={modal} variant="text" className={modalities[modal].style}>
                   {modalities[modal].name}
                 </Button>
               ))}
             </ButtonGroup>
-          </Grid>
-        </Grid>
+          </div>
+        </div>
       </CardContent>
     </Card>
-  )
+  );
 }
 
-function ResultContainer({result} : {result: Result[]}) {
+function ResultContainer({ result }: { result: Result[] }) {
   return (
-    <Stack spacing={2}>
-          {exampleResult.map((item) =>
-          <ResultCard key={item.dataset_uuid} nodeName={item.node_name} datasetName={item.dataset_name} datasetTotalSubjects={item.dataset_total_subjects} numMatchingSubjects={item.num_matching_subjects} imageModals={item.image_modals} />
-          )}
-    </Stack>
-  )
+    <div className="grid gap-4">
```

Comment:
this, I have to say, is nice. Such a simple and clear way to describe the layout